spikeinterface/comparison/addtemplatesrecording.py

```diff
@@ -0,0 +1,130 @@
+from typing import List, Union
+import numpy as np
+from spikeinterface.core import BaseRecording, BaseRecordingSegment, BaseSorting, BaseSortingSegment
+
+
+class AddTemplatesRecording(BaseRecording):
+    """
+    Class for creating a recording based on spike timings and templates.
+    Can be just the templates or can add to an already existing recording.
+
+    Parameters
+    ----------
+    sorting: BaseSorting
+        Sorting object containing all the units and their spike train.
+    templates: np.ndarray[n_units, n_samples, n_channels]
+        Array containing the templates to inject for all the units.
+    nbefore: list[int] | int | None
+        Where is the center of the template for each unit?
+        If None, will default to the highest peak.
```

Comment:
Maybe ms_before like we have in WavformExtractor could be better.
@alejoe91 ?

Reply:
I used `nbefore` following what's here: https://github.com/SpikeInterface/spikeinterface/blob/d75c2558c6e96a1077d27d00cac2ddab9b42797c/spikeinterface/core/waveform_extractor.py#L398